packages/sw-cache-expiration/test/sw/plugin.js

```diff
@@ -0,0 +1,226 @@
+  });
+
+  it(`should throw when Plugin() is called with an invalid maxAgeSeconds parameter`, function() {
+    let thrownError = null;
+    try {
+      new Plugin({maxAgeSeconds: 'invalid'});
+    } catch(err) {
+      thrownError = err;
+    }
+    expect(thrownError).to.exist;
+    expect(thrownError.name).to.equal('max-age-seconds-must-be-number');
+  });
+
+  it(`should use the maxAgeSeconds from the constructor`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    expect(plugin.maxAgeSeconds).to.equal(maxAgeSeconds);
+  });
+
+  it(`should use the maxEntries from the constructor`, function() {
+    const plugin = new Plugin({maxEntries});
+    expect(plugin.maxEntries).to.equal(maxEntries);
+  });
+
+  it(`should return the same IDB instance when getDB() is called multiple times`, function() {
+    const cacheName = getUniqueCacheName();
+    const plugin = new Plugin({maxEntries});
+    return plugin.getDB({cacheName}).then((firstDB) => {
+      return plugin.getDB({cacheName}).then((secondDB) => {
+        expect(firstDB).to.eql(secondDB);
+      });
+    });
+  });
+
+  it(`should return the same Cache instance when getCache() is called multiple times`, function() {
+    const cacheName = getUniqueCacheName();
+    const plugin = new Plugin({maxEntries});
+    return plugin.getCache({cacheName}).then((firstCache) => {
+      return plugin.getCache({cacheName}).then((secondCache) => {
+        expect(firstCache).to.eql(secondCache);
+      });
+    });
+  });
+
+  it(`should return true when there's no cachedResponse passed to isResponseFresh()`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    expect(plugin.isResponseFresh()).to.be.true;
+  });
+
+  it(`should return true when isResponseFresh() is called and there's no maxAgeSeconds`, function() {
+    const plugin = new Plugin({maxEntries});
+    const cachedResponse = new Response();
+    expect(plugin.isResponseFresh({cachedResponse})).to.be.true;
+  });
+
+  it(`should return true when isResponseFresh() is called and there's no Date: header in the cachedResponse`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    const cachedResponse = new Response();
+    expect(plugin.isResponseFresh({cachedResponse})).to.be.true;
+  });
+
+  it(`should return true when isResponseFresh() is called and the Date: header in the cachedResponse is recent`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    const date = new Date(now).toUTCString();
+    const cachedResponse = new Response('', {headers: {date}});
+    expect(plugin.isResponseFresh({cachedResponse, now})).to.be.true;
+  });
+
+  it(`should return false when isResponseFresh() is called and the Date: header in the cachedResponse is not recent`, function() {
+    const plugin = new Plugin({maxAgeSeconds});
+    // This will construct a date that is 1 second past the expiration.
+    const date = new Date(now - ((1 + maxAgeSeconds) * 1000)).toUTCString();
```

Comment:
Feel free to ignore this but worth considering:

You could condense some of these tests down into a single unit test (i.e. fresh with time as normal, then advance time and test again, still fresh, then advance past the maximum and ensure not fresh).

Using sinon fake timers you can control the time of the environment: (Example of using them here: https://github.com/gauntface/route/blob/master/packages/device-clock/test/node/device-clock.js#L17 and advance with https://github.com/gauntface/route/blob/master/packages/device-clock/test/node/device-clock.js#L46) Was thoroughly impressed how easy it was to use - no worries if you are happy leaving as is though.